diesel/src/pg/query_builder/query_fragment_impls.rs

```diff
@@ -98,15 +100,20 @@ where
     }
 }
 
-impl<T, U> QueryFragment<Pg, crate::pg::backend::PgOnConflictClaues>
-    for DecoratedConflictTarget<T, U>
+impl<Values, Target, Action, Where> QueryFragment<Pg, crate::pg::backend::PgOnConflictClaues>
+    for OnConflictValues<Values, DecoratedConflictTarget<Target, Where>, DoUpdate<Action>>
 where
-    T: QueryFragment<Pg>,
-    U: QueryFragment<Pg>,
+    Values: QueryFragment<Pg>,
+    Target: QueryFragment<Pg>,
+    DoUpdate<Action>: QueryFragment<Pg>,
+    Where: QueryFragment<Pg>,
```

Comment:
Curious about the change to what look like generics... Is this to make the code more readable? I've always been confused by the plethora of U, V, T, etc. letters all over the place :)

Reply:
Some of them are generics. Generics in rust must always be defined explicitly before usage. That happens by the `impl<Define, Generics, Here>` block at the beginning of the impl. For this case that means `Values`, `Target`, `Action` and `Where` are generics, anything else is a concrete type. 

As for what this change want's to achieve: It changes what queries are accepted or rejected at compile time. Before this change any generic action was accepted (by another impl than this one) in place of the last generic argument to `OnConflictValues`. This impl does not restrict that type parameter to the specific `DoUpdate<Action>` type. It's basically that which allows the added compile test to reject the invalid query now. That did not happen before.